### PR TITLE
=doc #1290 Merge common/marshalling.md and improve the Java version

### PR DESCRIPTION
**Pull request by jonas**
Refs #1290

### CODE REVIEW

**Comment by jonas** [comment]
Marked as WIP since I would like to also address #1367

Note that the Java marshaller API does not provide a way to use marshallers in a "stand-alone" fashion, so the `Using Marshallers` section probably doesn't make sense for Java.

File: docs/src/main/paradox/scala/http/common/marshalling.md [hunk 1]
```diff
@@ -124,10 +179,27 @@ depend on one being available implicitly at the usage site.
 
 ## Using Marshallers
 
-In many places through Akka HTTP, marshallers are used implicitly, e.g. when you define how to @ref[complete](../routing-dsl/directives/route-directives/complete.md) a
+In many places throughput Akka HTTP marshallers are used implicitly, e.g. when you define how to @ref[complete](../routing-dsl/directives/route-directives/complete.md) a
```
Comment (jonas) [inline docs/src/main/paradox/scala/http/common/marshalling.md]: Oops, should have been `throughout`.

File: akka-http/src/main/scala/akka/http/javadsl/marshalling/Marshaller.scala [hunk 2]
```diff
@@ -50,7 +50,9 @@ object Marshaller {
 
   def byteStringToEntity: Marshaller[ByteString, RequestEntity] = fromScala(marshalling.Marshaller.ByteStringMarshaller)
 
-  def fromDataToEntity: Marshaller[FormData, RequestEntity] = fromScala(marshalling.Marshaller.FormDataMarshaller)
+  @deprecated("Use `formDataToEntity` instead", "10.0.10")
+  def fromDataToEntity: Marshaller[FormData, RequestEntity] = formDataToEntity
```
Comment (raboof) [inline akka-http/src/main/scala/akka/http/javadsl/marshalling/Marshaller.scala]: ;)

File: docs/src/main/paradox/.htaccess [hunk 3]
```diff
@@ -2,5 +2,6 @@
 RedirectMatch 301 ^(.*)/common/de-coding.html$ $1/common/encoding.html
 RedirectMatch 301 ^(.*)/low-level-server-side-api.html$ $1/server-side/low-level-api.html
 RedirectMatch 301 ^(.*)/server-side-https-support.html$ $1/server-side/https-support.html
+RedirectMatch 301 ^(.*)/routing-dsl/marshalling.html$ $/common/marshalling.html
```
Comment (raboof) [inline docs/src/main/paradox/.htaccess]: missing a `1`

Comment (jrudolph) [reply docs/src/main/paradox/.htaccess]: Nice, didn't even know we had these...

**Comment by jonas** [comment]
Updated with comments, please review.

File: docs/src/main/paradox/scala/http/common/marshalling.md [hunk 4]
```diff
@@ -1,48 +1,61 @@
 # Marshalling
 
+@java[TODO @github[overhaul for Java](#1367)]
```
Comment (raboof) [inline docs/src/main/paradox/scala/http/common/marshalling.md]: 👍

File: docs/src/main/paradox/scala/http/common/marshalling.md [hunk 5]
```diff
@@ -124,10 +165,22 @@ depend on one being available implicitly at the usage site.
 
 ## Using Marshallers
 
-In many places through Akka HTTP, marshallers are used implicitly, e.g. when you define how to @ref[complete](../routing-dsl/directives/route-directives/complete.md) a
+In many places througout Akka HTTP, marshallers are used implicitly, e.g. when you define how to @ref[complete](../routing-dsl/directives/route-directives/complete.md) a
```
Comment (raboof) [inline docs/src/main/paradox/scala/http/common/marshalling.md]: throughout?

**Pull request merged**